### PR TITLE
Add model component

### PR DESCRIPTION
**Pull request by baering**

### CODE REVIEW

File: packages/tux/src/components/fields/Input.tsx [hunk 1]
```diff
@@ -3,20 +3,27 @@ import classNames from 'classnames'
...
+  constructor(props: Props) {
+    super(props)
+
+    const { value } = props
 
-  state = {
-    hasFocus: false
+    this.state = {
+      hasFocus: false
+    }
```
Comment (eirikurn) [inline packages/tux/src/components/fields/Input.tsx]: This constructor isn't doing anything, right?

File: packages/tux/src/components/fields/Input.tsx [hunk 2]
```diff
@@ -44,7 +51,7 @@ class Input extends Component<Props, State> {
           className="Input-field"
           id={id}
           onChange={(event) => onChange(event.target.value)}
-          value={value}
+          value={value ? value : DEFAULT_VALUE}
```
Comment (eirikurn) [inline packages/tux/src/components/fields/Input.tsx]: `value={value || ''}` would also work :)

File: packages/tux-adapter-contentful/src/locale.ts [hunk 3]
```diff
@@ -1,14 +1,26 @@
 export function extractLocale(model: any, locale: string) {
   for (const fieldName of Object.keys(model.fields)) {
     const fieldValue = model.fields[fieldName][locale]
-    model.fields[fieldName] = fieldValue
+    if (fieldValue) {
+      model.fields[fieldName] = fieldValue
+    } else {
+      model.fields[fieldName] = undefined
+    }
```
Comment (eirikurn) [inline packages/tux-adapter-contentful/src/locale.ts]: What was the issue with the existing syntax? Shouldn't it extract '' and 0 correctly?

File: packages/tux-adapter-contentful/src/locale.ts [hunk 4]
```diff
@@ -1,14 +1,26 @@
...
 export function injectLocale(model: any, locale: string) {
   for (const fieldName of Object.keys(model.fields)) {
-    const fieldValue = Object.assign(model.fields[fieldName])
+    let fieldValue
+    if (model.fields[fieldName]) {
+      if (model.fields instanceof Object) {
+        fieldValue = Object.assign(model.fields[fieldName])
+      } else {
+        fieldValue = model.fields[fieldName]
+      }
```
Comment (eirikurn) [inline packages/tux-adapter-contentful/src/locale.ts]: This condition is also weird. `model.fields` is always Object.

**Review by eirikurn — CHANGES_REQUESTED** [review]
Good first implementation. Some questions.

File: packages/tux/src/components/fields/Input.tsx (hunk 2 above)
Comment (baering) [reply packages/tux/src/components/fields/Input.tsx]: ~I like the idea of each component having a default value constant~

Ah, you meant `value || DEFAULT_VALUE`? I can change that, I just know that @rthor loves ternary so much.

File: packages/tux/src/components/fields/Input.tsx (hunk 1 above)
Comment (baering) [reply packages/tux/src/components/fields/Input.tsx]: Correct, forgot to remove - I think it served a purpose at some point but that was then refactored. Thanks.

File: packages/tux-adapter-contentful/src/locale.ts (hunk 3 above)
Comment (baering) [reply packages/tux-adapter-contentful/src/locale.ts]: I'll revert this.

File: packages/tux-adapter-contentful/src/locale.ts (hunk 4 above)
Comment (baering) [reply packages/tux-adapter-contentful/src/locale.ts]: This was missing a `model.fields[fieldName]`, I'll refactor this code.

File: packages/tux/src/components/fields/Input.tsx (hunk 2 above)
Comment (rthor) [reply packages/tux/src/components/fields/Input.tsx]: Ternaries are fine. I was quoting someone in that conversion... But sometimes they are really confusing.

File: packages/tux-adapter-contentful/src/locale.ts [hunk 5]
```diff
@@ -1,28 +1,28 @@
...
+    const fieldValueIsDefined = fieldValue !== undefined
+    const fieldValueIsObject = fieldValue instanceof Object
+
+    let injectedValue = undefined
+    if (fieldValueIsDefined) {
+      if (fieldValueIsObject) {
+        injectedValue = Object.assign(fieldValue)
       } else {
-        fieldValue = model.fields[fieldName]
+        injectedValue = fieldValue
```
Comment (eirikurn) [inline packages/tux-adapter-contentful/src/locale.ts]: The outer condition seems to be unnecessary.

**Review by eirikurn — APPROVED** [review]
Can you resolve conflicts and merge.

**Pull request merged**